### ISSUE TITLE
CloudFile.FetchAttributesAsync() fails to set ContentType on CoreCLR

### ISSUE THREAD

**Issue by chlowell** [Azure/azure-storage-net#227]
Using ```6.2.2-preview``` on runtime ```dnx-coreclr-win-x64.1.0.0-rc1-update1```. Here's a repro:

### Program.cs
```cs
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.File;
using System;
using System.IO;

namespace CloudFilePropertiesRepro
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var connectionString = "your secrets here";
            var fileToUpload = "Program.cs";

            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(connectionString);
            CloudFileClient fileClient = storageAccount.CreateCloudFileClient();

            CloudFileShare share = fileClient.GetShareReference("share");
            share.CreateIfNotExistsAsync().GetAwaiter().GetResult();

            CloudFileDirectory dir = share.GetRootDirectoryReference();
            dir.CreateIfNotExistsAsync().GetAwaiter().GetResult();

            CloudFile file = dir.GetFileReference("file");
            file.UploadFromFileAsync(fileToUpload, FileMode.OpenOrCreate).GetAwaiter().GetResult();

            Console.WriteLine("setting properties");
            file.Properties.ContentType = "for sure not null";
            file.SetPropertiesAsync().GetAwaiter().GetResult();
            Console.WriteLine($"ContentType is {file.Properties.ContentType}");

            Console.WriteLine("fetching attributes");
            file.FetchAttributesAsync().GetAwaiter().GetResult();
            Console.WriteLine($"ContentType is {file.Properties.ContentType ?? "null"}");
        }
    }
}
```
### project.json
```
{
  "frameworks": {
    "dnxcore50": {
      "dependencies": {
        "System.Console": "4.0.0-*",
        "WindowsAzure.Storage": "6.2.2-preview"
      }
    }
  },
  "compilationOptions": {
    "emitEntryPoint": true
  },
  "commands": {
    "CloudFilePropertiesRepro": "CloudFilePropertiesRepro"
  },
  "compile": [
    "**/*.cs"
  ]
}
```

### output
```
$ dnx run
setting properties
ContentType is for sure not null
fetching attributes
ContentType is null
```
The same code works as expected in a .NET 4.5 program using the ```net40``` assembly. ```SetPropertiesAsync()``` appears innocent: a file with ```ContentType``` set by the ```dnx``` program has the expected value in the .NET 4.5 program after ```FetchAttributesAsync()```.

**Comment by pemari-msft** [comment]
Sorry for the delay. We've seen something like this before with CoreCLR and proxies -- this may be a regression on their end. I'll take a look.

**Comment by erezvani1529** [comment]
Hi @chlowell, 

Update and a quick description of the problem:

Basically CoreCLR removes all ContentType values which do not follow the "type/subtype" format.
We were expecting this to be fixed with latest RC2 release and with it still present, we started a new issue and are following up with CoreCLR team dierctly: [https://github.com/dotnet/coreclr/issues/5806](url)

If you are blocked, you should be able to try a content-type value which is in "type/subtype" format and resolve your issue.

I will close this issue upon your response as it is being tracked with CoreCLR team.

Thanks!

**Comment by chlowell** [comment]
Thank you, I'm not blocked on this.

For future reference, this issue is tracked at dotnet/corefx#9433.
